### PR TITLE
CLI: Update renderer templates to provide correct typescript examples

### LINKED ISSUE #21179: [Bug]: HTML and Lit CLI templates for Typescript generate Javascript stories instead
### Describe the bug

I detected a [regression](https://github.com/storybookjs/sandboxes/commit/0cd7939bddd9eb3993d27c1592b65c0fd4d40560) in the sandbox repository:
image

Here are the templates which should contain ts stories:
[HTML-ts](https://github.com/storybookjs/sandboxes/tree/next/html-vite/default-ts/after-storybook/src/stories)
[Lit-ts](https://github.com/storybookjs/sandboxes/tree/next/lit-vite/default-ts/after-storybook/src/stories)

### To Reproduce

_No response_

### System

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by yannbf**
Closes #21179

## What I did

There was a change in the naming of directories inside of the CLI where `template/cli/ts` was split into `template/cli/ts-3-8` and `template/cli-ts-4-9`, but not all renderers had the directories renamed. This does that for:
- html
- web-components
- svelte

There's also Preact, which doesn't have typescript examples at all, but given it involves a little extra work (there's some conflicts with React errors), I'd say it could be done in a separate PR.

## How to test

1. Run a sandbox for template, e.g. `yarn task --task sandbox --start-from auto --template html-vite/default-ts`
2. Check that the generated stories are typescript

## Checklist

- [x] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [ ] Make sure to add/update documentation regarding your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [x] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

**Comment by yannbf** [comment]
@kasperpeulen @ndelangen given that we now provide a way to get templates from a community package, I wonder if the ts directory name change impacted them? Maybe we should also try and fallback to `cli/template/ts`?

**Comment by kasperpeulen** [comment]
Tagging @IanVS as he was involved in changing the template names

**Comment by IanVS** [comment]
Ah, yeah my [PR](https://github.com/storybookjs/storybook/pull/20458) started out to only handle ts-legacy templates, and it evolved from there so I guess I missed all the normal ts folders in renderers that didn't have ts-legacy.  

>we now provide a way to get templates from a community package

Can you point me to how that's documented?  Do we know of any packages doing this so far?

File: code/renderers/web-components/template/cli/ts-4-9/Header.stories.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  render: (args: HeaderProps) => Header(args),
+} satisfies Meta<HeaderProps>;
+
+export default meta;
+type Story = StoryObj<HeaderProps>;
+
+export const LoggedIn: Story = {
+  args: {
+    user: {
+      name: 'Jonh Doe',
```
Comment (IanVS) [inline code/renderers/web-components/template/cli/ts-4-9/Header.stories.ts:18]: typo?

**Review by IanVS — APPROVED** [review]
I was just wondering this morning why web-components didn't have TS in the sandbox.  Thanks for cleaning up after me here.

**Review by IanVS — APPROVED** [review]
I think we should be clear for anyone creating a community template that can be installed via the CLI what the correct template directory names should be, but I think it's also good to allow a fallback if they don't completely follow it.  @jonniebigodes do we have documentation for the community building new frameworks to install via our CL?  I'm not sure where I'd even look for that.

**Pull request merged**